README.md

```diff
@@ -1,10 +1,148 @@
 
-It uses Firebase Auth for persistent sessions and user DB as it's free and battle tested by millions of apps, but it's built in a modular way so other auth backends coudl be plugged in. 
+It uses Firebase Auth for persistent sessions and user DB as it's free and battle tested by millions of apps, but it's built in a modular way so other auth backends coudl be plugged in.
 
-**Project status**: this project is functionally complete but not yet nicely packaged with NPM and so on and is lacking docs. 
-We estimate to have that complete by Jan 23rd 2022
+> **Project status**: this project is functionally complete but not yet nicely packaged with NPM and so on and is lacking docs.
+> We estimate to have that complete by Jan 23rd 202
+
+CrossMint's solana-auth provides developers with the tools to configure their applications to allow users to authenticate with their Solana wallets.
+
+This package currently provides support for Firebase out of the box but allows developers to configure it to work with any backend/database.
+
+## 🚀 Quick Setup (Next.js Example)
+
+### Install
+
+```shell
+npx create-next-app crossmint-solana-auth --example "https://github.com/crossmint/solana-auth/packages/examples/nextjs-starter"
+
+cd crossmint-solana-auth && yarn install
+yarn dev
+```
+
+## 🛠 Setup (React + Firebase)
+
+## Install
+
+```sh
+yarn add @crossmint/solana-auth-base@beta  \
+         @crossmint/ solana-auth-react-ui@beta
+```
+
+Define `signIn` and `signOut`. Here we use firebase utils and `signInWithCustomToken`.
+
+```js
+// utils/firebaseClient.ts
+export async function signIn(data: Record<string, string>) {
+  const userCredential = await signInWithCustomToken(auth, data.token);
+  return userCredential;
+}
+
+export const signOut = () => {
+  return auth.signOut();
+};
+```
+
+Wrap your app in `SolanaAuthProvider`. Make sure to pass it a domain, your auth fucntions, and your serverless endpoints.
+
+```javascript
+// _app.tsx
+import { AppProps } from "next/app";
+import { FC } from "react";
+import { SolanaAuthProvider } from "../src";
+import { signIn, signOut } from "../utils/firebaseClient";
+
+import { SolanaAuthProvider } from "@crossmint/solana-auth-react-ui";
+
+require("../styles/globals.css");
+
+const App: FC<AppProps> = ({ Component, pageProps }) => {
+  return (
+    <SolanaAuthProvider
+      domain="example.xyz"
+      requestUrl="/api/solana-auth/getauthchallenge"
+      callbackUrl="/api/solana-auth/completeauthchallenge"
+      onAuthCallback={signIn}
+      signOut={signOut}
+    >
+      <Component {...pageProps} />
+    </SolanaAuthProvider>
+  );
+};
+
+export default App;
+```
+
+Use the `SolanaFirebaseAuth` component to render the signin button in your app.
```

Comment:
You say SolanaFirebaseAuth but then use SolanaAuthButtonFirebase

Which one is correct?